src/galileo/logger/logger.py

```diff
@@ -1455,22 +1473,66 @@ async def async_flush(self) -> list[Trace]:
         List[Trace]
             The list of uploaded workflows.
         """
+        if self.mode == "distributed":
+            return await self._flush_distributed()
         return await self._flush_batch()
 
-    async def _flush_batch(self) -> list[Trace]:
-        if self.mode == "distributed":
-            self._logger.warning("Flushing in distributed mode is not supported - traces are sent immediately.")
-            return []
+    def _conclude_trace_if_needed(self) -> None:
```

Comment:
Maybe a better name would be `_conclude_if_trace`? 

The only qualifier for the conclude being needed seems to be that the parent is a trace